server/historian/packages/historian-base/src/utils.ts

```diff
@@ -51,6 +51,43 @@ export function getTenantIdFromRequest(params: Params) {
     return "-";
 }
 
+export function getDocumentIdFromRequest(tenantId: string, authorization: string) {
+    if (!tenantId || tenantId === "-" || !authorization) {
+        return "-";
+    }
+
+    const token = parseToken(tenantId, authorization);
```

Comment:
Good point. As we discussed offline, refactoring the code to make `parseToken` happen at an upper layer and reuse it later (as opposed to what I'm doing, which is calling `parseToken` again) would be a complex task. So, for now, we can start by adding a try/catch to prevent the issue you described above.